src/main/java/book/store/service/shopping/cart/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package book.store.service.shopping.cart.impl;
+
+import book.store.dto.cart.item.CartItemRequestDto;
+import book.store.dto.cart.item.CartItemResponseDto;
+import book.store.dto.shopping.cart.ShoppingCartResponseDto;
+import book.store.exception.EntityNotFoundException;
+import book.store.mapper.CartItemMapper;
+import book.store.model.CartItem;
+import book.store.model.ShoppingCart;
+import book.store.model.User;
+import book.store.repository.shopping.cart.ShoppingCartRepository;
+import book.store.service.cart.item.CartItemService;
+import book.store.service.shopping.cart.ShoppingCartService;
+import java.util.HashSet;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private static final String CAN_T_FIND_SHOPPING_CART = "Can't find shopping cart!";
+
+    private final CartItemService cartItemService;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    public void register(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCart.setCartItems(new HashSet<>());
+        shoppingCartRepository.save(shoppingCart);
+
+    }
+
+    @Override
+    public CartItemResponseDto addCartItem(CartItemRequestDto requestDto,
+                                           Authentication authentication) {
+        ShoppingCart shoppingCart = getShoppingCart((User) authentication.getPrincipal());
```

Comment:
the logic should be 
find book
find shop cart
check if shop cart includes this book, if yes - increase quantity
if no - add cartItem to shop cart